src/components/Form/UserResourceForm.vue

```diff
@@ -0,0 +1,83 @@
+<template>
+  <v-sheet width="300" class="mx-auto">
+    <div class="mx-auto text-center" style="font-size: 24px">
+      {{ pageTitle }}
+    </div>
+
+    <v-form @submit.prevent="handleSubmit" ref="form" class="mt-4">
+      <v-select
+        label="Select user"
+        :items="userOptions"
+        v-model="formData.userOption"
+        :disabled="route.path.includes('/remove')"
+      >
+      </v-select>
+
+      <v-text-field
+        v-model="formData.quantity"
+        label="Quantity"
+        :rules="numberFieldRules"
+        required
+      ></v-text-field>
+
+      <div class="d-flex flex-column">
+        <v-btn color="success" class="mt-4" block type="submit">Submit</v-btn>
+        <v-btn color="error" class="mt-4" block @click="resetForm">Reset</v-btn>
+        <v-btn color="warning" class="mt-4" block @click="router.go(-1)"
+          >Go Back</v-btn
+        >
+      </div>
+    </v-form>
+  </v-sheet>
+</template>
+
+<script setup>
+import { useStore } from "vuex";
+import { ref, computed } from "vue";
+import { useNumberFieldRules } from "../../utils/validation-rules";
+import { useRoute, useRouter } from "vue-router";
+const store = useStore();
+const route = useRoute();
+const router = useRouter();
+const numberFieldRules = useNumberFieldRules();
+const formData = store.getters["resources/getResourceForm"];
+
+const userOptions = computed(() =>
+  store.getters["users/allUsers"].map((user) => user.name)
+);
+
+if (route.path.includes("/remove")) {
+  const resourceId = route.params.resourceId;
+  const loggedUser = route.params.userId;
+  const currentlyLoggedUser = computed(() => store.getters["users/getUserById"](loggedUser)).value
```

Comment:
These names are misleading. The userId is not necessarily the logged in user. It could be anyone. Please, change them accordingly.

Reply:
fixed